rfcs/0000-const-ub.md

```diff
@@ -21,43 +22,69 @@ There are some values that Rust needs to compute at compile-time.
 This includes the initial value of a `const`/`static`, and array lengths (and more general, const generics).
 Computing these initial values is called compile-time function evaluation (CTFE).
 CTFE in Rust is very powerful and permits running almost arbitrary Rust code.
-This begs the question, what happens when there is `unsafe` code and it causes Undefined Behavior (UB)?
+This begs the question, what happens when there is `unsafe` code and it causes [Undefined Behavior (UB)][UB]?
 
-The answer is that in this case, the final value that is currently being executed is arbitrary.
-For example, when UB arises while computing an array length, then the final array length can be any `usize`, or it can be (partially) uninitialized memory.
-No guarantees are made about this final value, and it can be different depending on host and target architecture, compiler flags, and more.
-However, UB will not otherwise adversely affect the currently running compiler; type-checking and lints and everything else will work correctly given whatever the result of the CTFE computation is.
-
-Note, however, that this means compile-time UB can later cause runtime UB when the program is actually executed:
-for example, if there is UB while computing the initial value of a `Vec<i32>`, the result might be a completely invalid vector that causes UB at runtime when used in the program.
-
-Sometimes, the compiler might be able to detect such problems and show an error or warning about CTFE computation having gone wrong (for example, the compiler might detect when the array length ends up being uninitialized).
-But other times, this might not be the case -- UB is not reliably detected during CTFE.
+The answer depends on the kind of UB: some kinds of UB are guaranteed to be detected,
+while other kinds of UB might either be detected, or else evaluation will continue as if the violated UB condition did not exist (i.e., as if this operation was actually defined).
 This can change from compiler version to compiler version: CTFE code that causes UB could build fine with one compiler and fail to build with another.
 (This is in accordance with the general policy that unsound code is not subject to strict stability guarantees.)
 
+[UB]: https://doc.rust-lang.org/reference/behavior-considered-undefined.html
+
 # Reference-level explanation
 [reference-level-explanation]: #reference-level-explanation
```

Comment:
As usual I have no idea how to split the RFC intention across "guide-level" and "reference-level" explanation.^^ That separation makes very little sense to me, for most of the RFCs that I end up writing...